rules/naming/src/Rector/Property/MakeBoolPropertyRespectIsHasWasMethodNamingRector.php

```diff
@@ -0,0 +1,216 @@
+    {
+        return new RectorDefinition('Renames property to respect is/has/was method naming', [
+            new CodeSample(
+                <<<'PHP'
+class SomeClass
+{
+    private $full = false;
+
+    public function isFull()
+    {
+        return $this->full;
+    }
+}
+PHP
+
+                ,
+                <<<'PHP'
+class SomeClass
+{
+    private $isFull = false;
+
+    public function isFull()
+    {
+        return $this->isFull;
+    }
+
+}
+PHP
+
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Property::class];
+    }
+
+    /**
+     * @param Property $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        $prefixedClassMethods = $this->getPrefixedClassMethods($node);
+        if ($prefixedClassMethods === []) {
+            return null;
+        }
+
+        $currentName = $this->getName($node);
+        $expectedName = $this->resolveExpectedName($prefixedClassMethods, $currentName);
+        if ($expectedName === null) {
+            return null;
+        }
+
+        $classLike = $node->getAttribute(AttributeKey::CLASS_NODE);
+        if ($classLike === null) {
+            return null;
+        }
+        $propertyProperty = $node->props[0];
+        $propertyProperty->name = new VarLikeIdentifier($expectedName);
+        $this->renamePropertyFetchesInClass($classLike, $currentName, $expectedName);
+
+        return $node;
+    }
+
+    private function renamePropertyFetchesInClass(ClassLike $classLike, string $oldName, string $expectedName): void
+    {
+        // 1. replace property fetch rename in whole class
+        $this->traverseNodesWithCallable([$classLike], function (Node $node) use (
+            $oldName,
+            $expectedName
+        ): ?PropertyFetch {
+            if (! $this->isLocalPropertyFetchNamed($node, $oldName)) {
+                return null;
+            }
+
+            /** @var PropertyFetch $node */
+            $node->name = new Identifier($expectedName);
+            return $node;
+        });
+    }
+
+    private function shouldSkip(Property $property): bool
+    {
+        if (! $property->isPrivate()) {
+            return true;
+        }
+
+        if (! $this->skipNonBooleanType($property)) {
+            return true;
+        }
+
+        if ($this->skipPrefixed($property)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private function skipNonBooleanType(Property $property): bool
+    {
+        /** @var PhpDocInfo|null $phpDocInfo */
+        $phpDocInfo = $property->getAttribute(AttributeKey::PHP_DOC_INFO);
+        if ($phpDocInfo === null) {
+            return false;
+        }
+
+        if (! $phpDocInfo->getVarType() instanceof BooleanType) {
+            return false;
+        }
+
+        return true;
```

Comment:
@TomasVotruba Is there any better way to figure out the property type? I guess, that current logic doesn't support PHP 7.4 property type. Also, I believe that it is possible to detect property type by value, not sure whether it is a good idea

Reply:
What about `$this->getStaticType()/getObjectType()` methods?

Reply:
If I remember correctly, I tried both and it didn't work but I'll give it a try tomorrow.

Reply:
I'd check other rules, that also use `$this->getStaticType()/getObjectType()` on `$property`.
There might be some tip/trick